### PR TITLE
add checkout flow

### PR DESCRIPTION
**Pull request by volodin28**
Selected items in basket go to checkout page.
User can add delivery details and tax address
As a result user should be assigned with unique order_id. Order details should be stored in DB,

### CODE REVIEW

File: src/apps/orders/views.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+                request,
+                self.template_name,
+                {
+                    "basket": basket,
+                    "total_price": total_price,
+                    "delivery_option_form": delivery_option_form,
+                },
+            )
+
+        else:
```
Comment (kleikoks) [inline src/apps/orders/views.py]: Move it to the top.
`if not request.user.is_authenticated:`
This way you can avoid unnecessary nesting

File: src/apps/orders/views.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+from src.apps.basket.basket import Basket
+from .forms import DeliveryOptionForm
+from .models import Order, OrderProduct
+
+
+class CheckoutView(View):
+    template_name = "orders/checkout.html"
+
+    def get(self, request, *args, **kwargs):
+        if request.user.is_authenticated:
```
Comment (egorvavilov) [inline src/apps/orders/views.py]: fix it with mixin pls

File: src/apps/orders/views.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+                self.template_name,
+                {
+                    "basket": basket,
+                    "total_price": total_price,
+                    "delivery_option_form": delivery_option_form,
+                },
+            )
+
+
+class ThankYouView(UserPassesTestMixin, View):
```
Comment (egorvavilov) [inline src/apps/orders/views.py]: Can we make it into order view? Model views has get_object method which is literally your custom get_order

**Review by egorvavilov — CHANGES_REQUESTED** [review]

File: src/apps/orders/views.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    context_object_name = "order"
+    pk_url_kwarg = "order_id"
+
+    def test_func(self):
+        order = self.get_object()
+        return self.request.user == order.user_id
+
+    def get(self, request, *args, **kwargs):
+        order = self.get_object()
+        print(order.id)
```
Comment (egorvavilov) [inline src/apps/orders/views.py]: oops, print statement

**Review by egorvavilov — CHANGES_REQUESTED** [review]

File: src/apps/orders/views.py (hunk 4 above)
Comment (volodin28) [reply src/apps/orders/views.py]: sry, was debugging and forgot to get rid of it

**Review by egorvavilov — APPROVED** [review]

**Pull request merged**